docs/caikit-servingruntime.yaml

```diff
@@ -1,26 +1,76 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: caikit-tgis-config
+data:
+  caikit-tgis.yml: |
+    runtime:
+      library: caikit_nlp
+      local_models_dir: /mnt/models/
+      lazy_load_local_models: true
+
+    model_management:
+      finders:
+        default:
+          type: MULTI
+          config:
+            finder_priority:
+              - tgis-auto
+        tgis-auto:
+          type: TGIS-AUTO
+          config:
+            test_connection: true
+      initializers:
+        default:
+          type: LOCAL
+          config:
+            backend_priority:
+              - type: TGIS
+                config:
+                  connection:
+                    hostname: localhost:8033
+---
 apiVersion: serving.kserve.io/v1alpha1
 kind: ServingRuntime
 metadata:
   name: caikit-runtime
 spec:
-  containers:
-  - env:
-    - name: RUNTIME_LOCAL_MODELS_DIR
-      value: /mnt/models
-    # TODO: This will eventually point to the official image
-    image: quay.io/opendatahub/caikit-tgis-serving@sha256:20cbc7b312e0dc63903d4a2c7d70b741135ffea31113af63dd8691da4baae0f9
-    name: kserve-container
-    ports:
-    # Note, KServe only allows a single port, this is the gRPC port. Subject to change in the future
-    - containerPort: 8085
-      name: h2c
-      protocol: TCP
-    resources:
-      requests:
-        cpu: 8
-        memory: 16Gi
   multiModel: false
-  supportedModelFormats:
-  # Note: this currently *only* supports caikit format models
-  - autoSelect: true
-    name: caikit
\ No newline at end of file
+  supportedModelFormats: # FIXME: check if caikit is the right value for supportedModelFormats
+    # Note: this currently *only* supports caikit format models
+    - autoSelect: true
+      name: caikit
+  containers:
+    - name: transformer-container
+      image: quay.io/opendatahub/text-generation-inference:stable-bafd218
+      command: ["text-generation-launcher"]
+      args: ["--model-name=/mnt/models/artifacts/"] # NOTE:--num-shard defaults to 1
+      env:
+        - name: TRANSFORMERS_CACHE
+          value: /tmp/transformers_cache
```

Comment:
For TGIS, I want to add the following into the template to align with what perf/scale has been testing, these are probably fine defaults:

```
    - name: RUNTIME_GRPC_SERVER_THREAD_POOL_SIZE
      value: "64" # This may need to scale higher, some models have this at 256, but this seems like a good default up from 5
    - name: MAX_BATCH_SIZE
      value: "256"
    - name: MAX_CONCURRENT_REQUESTS
      value: "64"
    - name: MAX_BATCH_WEIGHT
      value: "10000"
```

And a couple others that we may want to comment out, but leave in the SR for doc purposes

```
    - name: PT2_COMPILE
      value: "true" # This makes the model take a lot of time to load, but provides a speedup 
    - name: FLASH_ATTENTION
      value: "true" # This optimizes a few models that need it, should be commented by default with a link to https://github.com/IBM/text-generation-inference#converting-weights-to-safetensors-format
```

For sharding, we also need these fields, and *might* need to update the load_timeout in caikit, but I think that's likely not a problem in remote mode

Ref: https://github.com/IBM/text-generation-inference#running-sharded-models-tensor-parallel

```
    - name: NUM_GPUS
      value: "4" # Presumably however many they have
    - name: DEPLOYMENT_FRAMEWORK
      value: hf_custom_tp
```

It also pairs with the additional last line in the following caikit runtime-config.yaml:

```
    model_management:
      initializers:
        default:
          type: LOCAL
          config:
            backend_priority:
              - type: TGIS
                config:
                  local:
                    load_timeout: 2000
                    num_gpus: 4
```

Reply:
Given the potential size of the changes for sharding, it may be worth deferring that to a second SR spec that's set up specifically for it

Reply:
Also investigating if NUM_GPUs and similar need to be set explicitly or if there's detection logic

Reply:
Regarding sharding: I'd rather create a new manifest for it and include in another PR

Reply:
Regarding `max_batch_weight=10000`: it seems it is not a safe default, resulting in a tgis panic:

> thread 'main' panicked at 'max_batch_weight (10000) not large enough for max_sequence_length (2048)', /usr/src/router/src/server.rs:217:17

I'd comment this out for now, see #112 to reproduce

Reply:
Good catch, I'd also say it'd be good to have the others as (even commented) env variables, so users can override them in individual ISVCs without having to touch the entrypoint line there

Reply:
Yeah, overall, let's revert the cmdline args change, and just leave them as env vars commented so we can revisit in a subsequent PR, once that's done I'll merge this